packages/ember-htmlbars/tests/helpers/with_test.js

```diff
@@ -1,262 +1,13 @@
 import compile from 'ember-template-compiler/system/compile';
 import { runAppend, runDestroy } from 'ember-runtime/tests/utils';
 
 var view, lookup;
 var originalLookup = Ember.lookup;
 
-function testWithAs(moduleName, templateString, deprecated) {
-  QUnit.module(moduleName, {
-    setup() {
-      Ember.lookup = lookup = { Ember: Ember };
-
-      var template;
-      if (deprecated) {
-        expectDeprecation(function() {
-          template = compile(templateString);
-        }, 'Using {{with}} without block syntax (L1:C0) is deprecated. Please use standard block form (`{{#with foo as |bar|}}`) instead.');
-      } else {
-        template = compile(templateString);
-      }
-
-
-      view = EmberView.create({
-        template: template,
-        context: {
-          title: 'Señor Engineer',
-          person: { name: 'Tom Dale' }
-        }
-      });
-
-      runAppend(view);
-    },
-
-    teardown() {
-      runDestroy(view);
-      Ember.lookup = originalLookup;
-    }
-  });
-
-  QUnit.test('it should support #with-as syntax', function() {
-    equal(view.$().text(), 'Señor Engineer: Tom Dale', 'should be properly scoped');
-  });
-
-  QUnit.test('updating the context should update the alias', function() {
-    run(function() {
-      view.set('context.person', {
-        name: 'Yehuda Katz'
-      });
-    });
-
-    equal(view.$().text(), 'Señor Engineer: Yehuda Katz', 'should be properly scoped after updating');
-  });
-
-  QUnit.test('updating a property on the context should update the HTML', function() {
-    equal(view.$().text(), 'Señor Engineer: Tom Dale', 'precond - should be properly scoped after updating');
-
-    run(function() {
-      set(view, 'context.person.name', 'Yehuda Katz');
-    });
-
-    equal(view.$().text(), 'Señor Engineer: Yehuda Katz', 'should be properly scoped after updating');
-  });
-
-  QUnit.test('updating a property on the view should update the HTML', function() {
-    run(function() {
-      view.set('context.title', 'Señorette Engineer');
-    });
-
-    equal(view.$().text(), 'Señorette Engineer: Tom Dale', 'should be properly scoped after updating');
-  });
-}
-
-QUnit.module('Multiple Handlebars {{with foo as |bar|}} helpers', {
-  setup() {
-    Ember.lookup = lookup = { Ember: Ember };
-  },
-
-  teardown() {
-    runDestroy(view);
-
-    Ember.lookup = originalLookup;
-  }
-});
-
-QUnit.test('re-using the same variable with different #with blocks does not override each other', function() {
-  view = EmberView.create({
-    template: compile('Admin: {{#with admin as |person|}}{{person.name}}{{/with}} User: {{#with user as |person|}}{{person.name}}{{/with}}'),
-    context: {
-      admin: { name: 'Tom Dale' },
-      user: { name: 'Yehuda Katz' }
-    }
-  });
-
-  runAppend(view);
-  equal(view.$().text(), 'Admin: Tom Dale User: Yehuda Katz', 'should be properly scoped');
-});
-
-QUnit.test('should respect `isTruthy` field on a view', function() {
```

Comment:
Actually, I think we removed coverage for this – this test is about `isTruthy` beging set on the `self` context directly. I don't know why you would want to do it, and I doubt that it is super important, but it is also not hard to do it generically, so we should probably maintain it:

```javascript
export class SharedConditionalsTest extends AbstractConditionalsTest {

  ...

  ['@test it tests for `isTruthy` on the context if available']() {
    let template = this.templateFor({ cond: 'this', truthy: 'T1', falsy: 'F1' });

    this.render(template, { isTruthy: true });

    this.assertText('T1');

    this.runTask(() => this.rerender());

    this.assertText('T1');

    this.runTask(() => set(this.context, 'isTruthy', false));

    this.assertText('F1');

    this.runTask(() => set(this.context, 'isTruthy', true));

    this.assertText('T1');
  }

  ...

}
```